Library/Homebrew/requirements/gpg_requirement.rb

```diff
@@ -0,0 +1,29 @@
+require "requirement"
+
+class GPGRequirement < Requirement
+  fatal true
+  default_formula "gnupg2"
+
+  satisfy(:build_env => false) { gpg2 || gpg1 }
+
+  # Homebrew installs GnuPG 1.x as `gpg` but this may change in future.
+  # MacGPG2/GPGTools installs GnuPG 2.0.x as a vanilla `gpg` symlink
+  # pointing to `gpg2`. Ensure we're not using unexpected gpg version.
+  def gpg1
+    which_all("gpg").detect do |gpg|
+      gpg_version = Utils.popen_read(gpg, "--version")[/\d\.\d/, 0]
+      next unless gpg_version
+      Version.new(gpg_version.to_s) == Version.new("1.4")
+    end
+  end
+
+  def gpg2
+    which_all("gpg2").detect do |gpg2|
+      gpg2_version = Utils.popen_read(gpg2, "--version")[/\d\.\d/, 0]
+      next unless gpg2_version
+      # For now, to give upstreams time to adjust, only support 2.0.x
+      # rather than the 2.1.x "modern" series.
+      Version.new(gpg2_version.to_s) == Version.new("2.0")
+    end
+  end
```

Comment:
> Maybe call the variable gpg_short_version.

Sure, happy to do that.

> It fails to consider future 2.2+ release?

That's a monster hypothetical, to be honest. GPG 1.x last changed the x in 2004, GPG 2.x started as 2.0 in 2006 & the first release of 2.1 which is an entirely separate branch was cut at the end of 2014.

Despite "2.0" and "2.1" seeming fairly related they are enormously different releases & upstream consider them unique branches. 2.2 would be the same situation, unless upstream decide to drop the versioning scheme they've used for 15+ years now.

> What about when upstream fixes the issue?

If you mean GnuPG, it won't be fixed. 2.1 was intentionally breaking for the sake of a less fragmented codebase & introduction of new features.